packages/quick_actions/quick_actions_ios/ios/Classes/ShortcutStateManaging.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import UIKit
+
+/// Manages the shortcut related states.
+protocol ShortcutStateManaging {
```

Comment:
Does it make sense for there to be a `var shortcutItems: [UIApplicationShortcutItem]? { get set }` protocol as well as this one?  It looks like `DefaultShortcutStateManager` could easily implement `ShortcutItemServicing` instead by exposing `service.shortcutItems`.

Reply:
The `var shortcutItems: [UIApplicationShortcutItem]? { get set }` protocol is just an abstraction to stub `UIApplication`. The ShortcutStateManager has extra parsing logic. So it is better not to combine these 2. 

An alternative way is to define a static `ShortcutItemParser`, like this: 

```
protocol ShortcutItemParser {
  static func parseItems(_ items: [[String, Any]]) -> [UIApplicationShortcutItem]
```

This will make the dependency DAG more "flat". Let me try and see how it goes.

Reply:
I updated `ShortcutStateManager` to `ShortcutItemParser`, and renamed `ShortcutItemServicing` to `AppShortcutControlling`. 

The benefit is simpler and more flat dependency tree: Now the `QuickActionsPlugin` depends on both `ShortcutItemParser` and `AppShortcutControlling`, and these 2 subcomponents do not depend on each other anymore. 

PTAL thx

Reply:
Note that we still want to have a separate `AppShortcutControlling` protocol, so that we can stub out the `UIApplication.default`

Reply:
`ShortcutItemParser` SGTM.

But `AppShortcutControlling` isn't a capability in the sense the docs mean, in that setting this `shortcutItems` collection is not operating on the "implementing" `UIApplication`, it's more mutating it.  Like the example `ProgressReporting` would be reporting on the progress of the thing implementing the protocol.

In addition, "AppShortcutControlling" doesn't exactly make sense in English.  How about `ShortcutItemProvider`?  At the end of the day Swift's naming guidance is purposely very loosey, we should name it in a way that describes what it does, and also makes sense in English.

Reply:
Hmmm, `Provide` sounds like it only provides something, with only getter and no setters (e.g. `AgeProviding` in the style guide). 

How about `AppShortcutManaging`?

Reply:
i'm circling back on this - i think i am fine with the `Provider` name (it's not ideal, as explained above), but i can't think of a better name. 

Though we want `Providing` for protocol name, since it's a capability to provide something (similar to `NameProviding` and `AgeProviding` [in our style guide](https://google.github.io/swift/). What do you think?

Reply:
Sounds good! Thanks for iterating on this with me!

Reply:
Renamed to `ShortcutItemProviding` (unless @cyanglaz or @stuartmorgan can think of a better name?)